app/models/user.rb

```diff
@@ -41,15 +45,23 @@ def self.search(term)
   end
 
   def avatar
-    twitch.avatar
+    if twitch.present?
```

Comment:
With this and URI, as more login methods are added these checks seem like they would get out of hand a bit quickly.